### PR TITLE
Update workflows and dependencies

### PR DESCRIPTION
**Pull request by CDzungx**
Update things for no reason
 - Update workflows deps
 - Update rust version
 - Update package.json
 - Update Base64, sysinfo
 - Fix clippy err backend mac-os
 
Tested on Windows 11 and see no diffirent

### CODE REVIEW

File: src-tauri/src/functionality/rpc.rs [hunk 1]
```diff
@@ -54,11 +54,11 @@ pub fn append_to_local(detectables: Vec<DetectableActivity>) {
...
 pub fn start_rpc_server(win: tauri::Window) {
   let detectable = reqwest::blocking::get(
-    "https://gist.githubusercontent.com/SpikeHD/209bd9b17c97f45dc5be4803c748726f/raw/ddf8ed33621933b4e3c58cf1113e1679ab9fd9b5/dorion_detectable.json",
-  )
-  .expect("Request for detectable.json failed")
-  .text()
-  .expect("Failed to get text from response");
+        "https://gist.githubusercontent.com/SpikeHD/209bd9b17c97f45dc5be4803c748726f/raw/ddf8ed33621933b4e3c58cf1113e1679ab9fd9b5/dorion_detectable.json",
+    )
+        .expect("Request for detectable.json failed")
```
Comment (SpikeHD) [inline src-tauri/src/functionality/rpc.rs:59]: This never generated a formatting error before, does it now? If not pls leave as-is :)

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -84,11 +84,11 @@ custom-protocol = ["tauri/custom-protocol"]
 
 [target.x86_64-unknown-linux-gnu]
 rustflags = [
-  "-C", "link-arg=-fuse-ld=lld",
+    "-C", "link-arg=-fuse-ld=lld",
```
Comment (SpikeHD) [inline src-tauri/Cargo.toml:87]: I know this is horrifically and annoyingly nitpicky but this indentation should stay the same (unless it makes fmt mad now?)

File: src-tauri/src/processors/css_preprocess.rs [hunk 3]
```diff
@@ -201,13 +203,13 @@ pub async fn localize_images(win: tauri::Window, css: String) -> String {
...
     // CORS allows discord media
     if url.is_empty()
-      || url.contains(".css")
-      || url.contains("data:image")
-      || url.contains("media.discordapp")
-      || url.contains("cdn.discordapp")
-      || url.contains("discord.com/assets")
-      // Imgur is allowed(?)
-      || url.contains("i.imgur.com")
+            || url.contains(".css")
```
Comment (SpikeHD) [inline src-tauri/src/processors/css_preprocess.rs:206]: All of this indentation should also stay the same unless for some reason it's a problem now

File: src-tauri/Cargo.toml (hunk 2 above)
Comment (CDzungx) [reply src-tauri/Cargo.toml:87]: Idk if cargo fmt format this or my editor did this 🤔

**Comment by CDzungx** [comment]
You can just ignore if youre not sure 😂 about the indent, it's cargo fmt not my editor.
The clone is clippy suggestion.
The base64 is migrated as [this guide](https://docs.rs/base64/latest/base64/engine/trait.Engine.html#method.encode)

**Comment by SpikeHD** [comment]
Ahaha alr no problem then, thanks for the PR!

**Pull request merged**